From ffadb1f71745289497ca4a786a368d0a1edd18b7 Mon Sep 17 00:00:00 2001
From: Tim Starling <tstarling@wikimedia.org>
Date: Wed, 26 Nov 2014 11:09:32 +1100
Subject: [PATCH 1/2] Don't install bundled PCRE

Don't install the bundled libpcre, since it is harmful and unnecessary
for the same reasons as installation of libzip, which I removed in pull
request #16.

Don't compile the bundled PCRE or libzip if they are not going to
be used. HPHPSetup has been included at the relevant time, so we
already have this information. If the bundled libraries are not used, we
don't want to install header files for them, since this will could cause
errors in DSO extensions if the system has a different library version.
Plus, it helps the build go faster if you don't compile unnecessary
things.
---
 CMakeLists.txt      | 14 +++++++++++---
 pcre/CMakeLists.txt | 32 --------------------------------
 2 files changed, 11 insertions(+), 35 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 71a9326..9b5df6a 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -24,7 +24,6 @@ set(THIRD_PARTY_MODULES)
 set(THIRD_PARTY_HEADERS)
 
 list(APPEND THIRD_PARTY_MODULES
-  pcre
   fastlz
   libafdt
   libmbfl
@@ -32,13 +31,22 @@ list(APPEND THIRD_PARTY_MODULES
   timelib
   lz4
   double-conversion
-  folly
-  libzip)
+  folly)
 if(ENABLE_FASTCGI)
   list(APPEND THIRD_PARTY_MODULES proxygen)
   list(APPEND THIRD_PARTY_MODULES thrift)
 endif()
 
+# Add bundled PCRE if the system one will not be used
+if (NOT PCRE_LIBRARY)
+  list(APPEND THIRD_PARTY_MODULES pcre)
+endif()
+
+# Add bundled libzip if the system one will not be used
+if (NOT LIBZIP_LIBRARY)
+  list(APPEND THIRD_PARTY_MODULES libzip)
+endif()
+
 function(TP_INSTALL_HEADERS TARGET SRCDIR DEST)
   file(GLOB_RECURSE files "${SRCDIR}/*.h")
   install(
